DylibBinder/DBAssociatedType.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using SwiftReflector;
+
+namespace DylibBinder {
+	internal class DBAssociatedType {
+		public string Name { get; }
+
+		public DBAssociatedType (string name)
+		{
+			if (string.IsNullOrEmpty (name))
+				throw new ArgumentException ("DBAssociateType name cannot be null or empty");
+			Name = name;
+		}
+	}
+
+	internal class DBAssociatedTypes {
+		public HashSet<DBAssociatedType> AssociatedTypeCollection { get; } = new HashSet<DBAssociatedType> (new DBAssociatedTypeComparer ());
+
+		public DBAssociatedTypes ()
+		{
+		}
+
+		public DBAssociatedTypes (DBTypeDeclaration typeDeclaration)
+		{
+			AssociatedTypeCollection.UnionWith (typeDeclaration.GetAssociatedTypes ());
+		}
+	}
+
+	// interface for having an AssociatedTypes Property
+	internal interface IAssociatedTypes {
+		public DBAssociatedTypes AssociatedTypes { get; }
+	}
+
+	internal static class DBAssociatedTypesExtensions {
+		public static List<DBAssociatedType> GetAssociatedTypes (this SwiftBaseFunctionType signature)
+			=> signature.EachParameter.SelectMany (t => AssociatedTypesSwitch (t)).ToList ();
```

Comment:
I think we can do INeumerables in all this methods, create an issue to do that and get a little better. No need for this PR, please reply with the issue to this comment.

Reply:
https://github.com/xamarin/binding-tools-for-swift/issues/686